### DISCUSSION TITLE
About motion file

### DISCUSSION THREAD

**Discussion by WXY-0324** [cyberbotics/wrestling#71] [Q&A] [answered]
Hi
Recently, I was confronted with the questions about .motion file because of not knowing how to create my personal motion.
And I tried to use Choregraphe, but I found that it is very tricky, and I have no idea about how to extract the .motion file from Choregraphe.
is there something available for me to design my personal motion?
I would really appreciate it if you could help me address this problem

**Comment by omichel** [comment, marked as answer]
Unfortunately, there is no easy-to-use tool that I can recommend to create your own motion files.
Choregraphe is not a good option, as it doesn't work any more with Webots.
However, motion files are human readable text files which I believe are fairly easy to understand.
For example, if you look into the `HandWave.motion` file, you will see:
```csv
#WEBOTS_MOTION,V1.0,RShoulderPitch,RShoulderRoll,LShoulderPitch
00:00:000,Pose1,-1.5,0,1.49797
00:00:040,Pose2,-1.5,0.1,*
00:00:080,Pose3,-1.5,0.2,*
00:00:120,Pose4,-1.5,0.3,*
00:00:160,Pose5,-1.5,0.2,*
00:00:200,Pose6,-1.5,0.1,*
00:00:240,Pose7,-1.5,0,*
00:00:280,Pose8,-1.5,-0.1,*
00:00:320,Pose9,-1.5,-0.2,*
00:00:360,Pose10,-1.5,-0.3,*
00:00:400,Pose11,-1.5,-0.4,*
00:00:440,Pose12,-1.5,-0.5,*
00:00:480,Pose13,-1.5,-0.5,*
00:00:520,Pose14,-1.5,-0.4,*
00:00:560,Pose15,-1.5,-0.3,*
00:00:600,Pose16,-1.5,-0.2,*
00:00:640,Pose17,-1.5,-0.1,*
00:00:680,Pose18,-1.5,0,*
00:00:720,Pose19,-1.5,0.1,*
00:00:760,Pose20,-1.5,0.2,*
...
```
Basically, for each line, the first value is the time stamp, the second value is the name of the pose and the following values are the motor target positions. In that case the motion file controls only 3 motors: `RShoulderPitch`, `RShoulderRoll` and `LShoulderPitch`. The star (`*`), means that the target value is not set (and hence can be controlled directly using the Webots motor API while the motion is playing). To create your own tool that produce motion files, you can use set the robot in some position and print the position of its motors into a file.

**Reply by JustLuckyRobotics** [reply]
I'd also add that for making custom motions with webots its advantageous to make a separate version of your participant code that   essentially only calls your custom motion file. doing this allows you to simply change values -> save-> restart sim and evaluate. helps a lot with developing custom motions in isolation.

**Reply by WXY-0324** [reply]
Got it! Thank you!!
